easybuild/tools/module_generator.py

```diff
@@ -238,18 +285,181 @@ def set_alias(self, key, value):
+             '  load("%(name)s/%(version)s")',
+             'end',
+             ])
+
+        elif conflict:
+            # conflicts are not needed in lua module files, as Lmod's one name
+            # rule and automatic swapping.
+            pass
+
+        txt = '\n'.join(lines) % {
+            'name': self.app.name,
+            'version': self.app.version,
+            'description': description,
+            'installdir': self.app.installdir,
+            'homepage': self.app.cfg['homepage'],
+        }
+
+        return txt
+
+    def load_module(self, mod_name):
+        """
+        Generate load statements for module.
+        """
+        if build_option('recursive_mod_unload'):
+            # not wrapping the 'module load' with an is-loaded guard ensures recursive unloading;
+            # when "module unload" is called on the module in which the depedency "module load" is present,
+            # it will get translated to "module unload"
+            load_statement = ['load("%(mod_name)s")']
         else:
-            self.module_path = config.install_path('mod')
+            load_statement = [
+                'if ( not isloaded("%(mod_name)s")) then',
+                '  load("%(mod_name)s")',
+                'end',
+            ]
+        return '\n'.join([""] + load_statement + [""]) % {'mod_name': mod_name}
 
-    def is_fake(self):
-        """Return whether this ModuleGenerator instance generates fake modules or not."""
-        return self.fake
+    def unload_module(self, mod_name):
+        """
+        Generate unload statements for module.
+        """
+        return '\n'.join([
+            "",
+            "if (isloaded(%(mod_name)s)) then",
+            "    unload(%(mod_name)s)",
+            "end",
+            "",
+        ]) % {'mod_name': mod_name}
+
+    def prepend_paths(self, key, paths, allow_abs=False):
+        """
+        Generate prepend-path statements for the given list of paths.
+        """
+        template = 'prepend_path(%s,%s)\n'
+
+        if isinstance(paths, basestring):
+            self.log.info("Wrapping %s into a list before using it to prepend path %s" % (paths, key))
+            paths = [paths]
+
+        # make sure only relative paths are passed
+        for i in xrange(len(paths)):
+            if os.path.isabs(paths[i]) and not allow_abs:
+                self.log.error("Absolute path %s passed to prepend_paths which only expects relative paths." % paths[i])
+            elif not os.path.isabs(paths[i]):
+                # prepend $root (= installdir) for relative paths
+                paths[i] = ' pathJoin(pkg.root,"%s")' % paths[i]
+
+        statements = [template % (quote_str(key), p) for p in paths]
+        return ''.join(statements)
+
+    def use(self, paths):
+        """
+        Generate module use statements for given list of module paths.
+        """
+        use_statements = []
+        for path in paths:
```

Comment:
yes, if not, there's a bug in the framework

@pforai: just document this in the docstring

```python
@param paths: list of module path extensions to generate use statements for
```